my_enumerables.rb

```diff
@@ -0,0 +1,203 @@
+module Enumerable
+  def my_each
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << (yield self[i])
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < to_a.length
+        modified_object << (yield self[i], i)
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << self[i] if (yield self[i]) == true
+        i += 1
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length == length
+  end
+
+  def my_any?(pattern = nil)
+    false if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.positive?
+  end
+
+  def my_none?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length.zero?
+  end
+
+  def my_count(*args)
+    count_array = []
+
+    i = 0
+    if block_given?
+      while i < length
+        count_array << self[i] if yield self[i]
+        i += 1
+      end
+    elsif !block_given? && !args.empty?
+      while i < length
+        count_array << self[i] if self[i] == args.first
+        i += 1
+      end
+    else
+      count_array = self
+    end
+    count_array.length
+  end
+
+  def my_map
+    object = to_a
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < to_a.length
+        modified_object << (yield object[i])
+        i += 1
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
```

Comment:
`my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
```ruby
my_proc = Proc.new { |num| num > 10 }
[11, 2, 3, 15].my_map(my_proc) {|num| num < 10 } # should return [true, false, false, true]
```
![image](https://user-images.githubusercontent.com/38625956/84071501-a04e8600-a9ef-11ea-88c2-e74e1b0c15e4.png)